airflow/utils/db.py

```diff
@@ -854,7 +854,7 @@ def check_conn_id_duplicates(session: Session) -> Iterable[str]:
         )
 
 
-def reflect_tables(tables: List[Union[Base, str]], session):
+def reflect_tables(tables: Optional[List[Union[Base, str]]], session):
```

Comment:
Why is this `None` case needed? From what I can tell it’s the same as `reflect_tables(tables=[])`. Using `None` is probably more straightforward at the call site, but in that case we should change the condition in this function to `if tables is None` instead.

Reply:
Right exactly, calling `reflect_tables(tables=[])` feels just a little bit icky compared to `reflect_tables(tables=None)`.

I guess the idea with `if not tables:` is that whether user gives `[]` or `None` the effect is the same -- we'll reflect all tables.  But I guess yeah we could make it so that if you pass empty list it just does _nothing_  -- just seems odd, why you'd want to do that...

Reply:
OK I guess i can see the case for `[]` means no tables.

Consistent behavior.  e.g. in the dynamic case:
```
metadata = reflect_tables([x[0] for x in models_list], session)
```

well maybe there you don't need to bother reflecting any tables.

updated.